src/mon/OSDMonitor.cc

```diff
@@ -8598,9 +8598,8 @@ bool OSDMonitor::prepare_command_impl(MonOpRequestRef op,
      CrushWrapper newcrush;
      _get_pending_crush(newcrush);
      if (!newcrush.class_exists(device_class)) {
-       err = -ENOENT;
-       ss << "class '" << device_class << "' does not exist";
-       goto reply;
+       err = 0; // make command idempotent
+       goto wait;
```

Comment:
if you goto wait here, you *also* need
```
if (!crush.class_exists(device_class)) {
￼        err = -ENOENT;
￼        ss << "class '" << device_class << "' does not exist";
￼        goto reply;
    }
```
a bit further up

Reply:
@liewegas  Do you mean:

```
if (!crush.class_exists(device_class)) {
￼        err = 0;
￼        ss << "class '" << device_class << "' does not exist";
￼        goto reply;
    }
```
?